scripts/update_site_stats.py

```
@@ -0,0 +1,203 @@
+    return github_url_request(f"https://api.github.com{path}{query}", token)
+
+
+def github_url_request(url: str, token: str | None = None) -> tuple[object, object]:
+    request = urllib.request.Request(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "User-Agent": "aidevops.sh-site-stats",
+            **({"Authorization": f"Bearer {token}"} if token else {}),
+        },
+    )
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)